### PR TITLE
MV-7023 - add api docs for shortcodes.

### PR DESCRIPTION
**Pull request by maksymmateiuk**
**⚠️Please do not make any changes to spec files in this repository!⚠️**

**All spec changes should be done in the [api-specs](https://github.com/Bandwidth/api-specs) repository.**

**Please follow the [Guide](https://bandwidth-jira.atlassian.net/wiki/spaces/DX/pages/4098359409/How+to+Update+API+Specifications+and+Contribute+to+Developer+Documentation) for contributing to our developer documentation.**

### CODE REVIEW

File: site/specs-temp/numbers.json [hunk 1]
```diff
@@ -30400,6 +30496,133 @@
...
+              "RANDOM",
+              "VANITY"
+            ]
+          },
+          "CustomerManagedLease": {
+            "type": "boolean"
+          }
+        }
+      },
+      "ShortCodeResponseList": {
```
Comment (ckoegel) [inline site/specs-temp/numbers.json:30570]:
```suggestion
      "shortCodeResponseList": {
```
like Brian mentioned, there are a lot of uncaught camelCase issues in this spec, but going forward we should try to make these schema names conform to the standard.

File: site/specs-temp/numbers.json [hunk 2]
```diff
@@ -12901,6 +12901,102 @@
...
+        "description": "Retrieve shortcodes",
+        "summary": "Retrieve shortcodes",
+        "operationId": "RetrieveShortcodes",
+        "parameters": [
+          {
+            "$ref": "#/components/parameters/AccountIdPathParam"
+          }
+        ],
+        "responses": {
+          "200": {
```
[the remark this answers is not in the document]
Comment (ckoegel) [reply site/specs-temp/numbers.json:12918]: Additionally, are there going to be other possible response codes from this and the other endpoints (400, 404, etc.)? If so, they should be documented as well

File: site/specs-temp/numbers.json [hunk 3]
```diff
@@ -30400,6 +30496,133 @@
...
+          },
+          "LeaseStartDate": {
+            "type": "string"
+          },
+          "LeaseEndDate": {
+            "type": "string"
+          },
+          "Status": {
+            "type": "string",
+            "enum": [
```
[the remark this answers is not in the document]
Comment (maksymmateiuk) [reply site/specs-temp/numbers.json]: We can't. Iris already has Status enum with different values and if we will try add all Statuses enum there will be duplication.

File: site/specs-temp/numbers.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (maksymmateiuk) [reply site/specs-temp/numbers.json]: How can we reuse responses with different examples?

File: site/specs-temp/numbers.json [hunk 4]
```diff
@@ -13352,6 +13352,103 @@
...
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/accounts/{accountId}/shortcodes/{shortcode}": {
```
Comment (sarmstrong-bw) [inline site/specs-temp/numbers.json:13387]: aren't we changing this to include the country?

Comment (sarmstrong-bw) [reply site/specs-temp/numbers.json:13387]:
```suggestion
    "/accounts/{accountId}/shortcodes/{shortcode}/{country}": {
```

File: site/specs-temp/numbers.json [hunk 5]
```diff
@@ -13352,6 +13352,103 @@
...
+        "parameters": [
+          {
+            "$ref": "#/components/parameters/AccountIdPathParam"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Retrieve list shortcodes",
+            "content": {
+              "application/xml": {
```
Comment (brianluisgomez) [inline site/specs-temp/numbers.json:13372]: Revise to only return application/json responses

File: site/specs-temp/numbers.json [hunk 6]
```diff
@@ -13352,6 +13352,103 @@
...
+        "description": "Retrieve shortcode",
+        "summary": "Retrieve shortcode",
+        "operationId": "getShortcode",
+        "parameters": [
+          {
+            "$ref": "#/components/parameters/AccountIdPathParam"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (brianluisgomez) [inline site/specs-temp/numbers.json:13401]: Include 202 response in case provisioning isn't completed.

Comment (maksymmateiuk) [reply site/specs-temp/numbers.json:13401]: Need a description for 202.

File: site/specs-temp/numbers.json [hunk 7]
```diff
@@ -13430,6 +13430,169 @@
...
+          "403": {
+            "$ref": "#/components/responses/forbiddenError"
+          },
+          "404": {
+            "$ref": "#/components/responses/notFoundError"
+          },
+          "405": {
+            "$ref": "#/components/responses/notAllowedError"
+          },
+          "409": {
```
Comment (nirmitpatel) [inline site/specs-temp/numbers.json:13584]: We won't have 409 getting returned for method

File: site/specs-temp/numbers.json [hunk 8]
```diff
@@ -13430,6 +13430,169 @@
...
```
Comment (nirmitpatel) [inline site/specs-temp/numbers.json:13533]: We won't have 409 getting returned for GET morthods

File: site/specs-temp/numbers.json [hunk 9]
```diff
@@ -13430,6 +13430,169 @@
...
+                }
+              }
+            }
+          },
+          "202": {
+            "description": "Retrieve shortcode",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/shortCodeResponse"
```
Comment (nirmitpatel) [inline site/specs-temp/numbers.json:13513]: I think 202 will return a different body like prob a message that says "Order already in progress" or something like that. We will groom a story for it and will let you know.

File: site/specs-temp/numbers.json [hunk 10]
```diff
@@ -13430,6 +13430,169 @@
...
+      "get": {
+        "tags": [
+          "Short codes"
+        ],
+        "description": "Retrieve shortcode",
+        "summary": "Retrieve shortcode",
+        "operationId": "getShortcode",
+        "parameters": [
+          {
+            "$ref": "#/components/parameters/AccountIdPathParam"
```
Comment (nirmitpatel) [inline site/specs-temp/numbers.json:13494]: It will take three params - accountID, shortcode and country

File: site/specs-temp/numbers.json [hunk 11]
```diff
@@ -13430,6 +13430,169 @@
...
+          "403": {
+            "$ref": "#/components/responses/forbiddenError"
+          },
+          "404": {
+            "$ref": "#/components/responses/notFoundError"
+          },
+          "405": {
+            "$ref": "#/components/responses/notAllowedError"
+          },
+          "409": {
```
Comment (nirmitpatel) [inline site/specs-temp/numbers.json:13472]: No 409 for GET

File: site/specs-temp/numbers.json (hunk 9 above)
Comment (maksymmateiuk) [reply site/specs-temp/numbers.json:13513]: Yeah, I just made 202 - like a mock. Thank you!

**Pull request closed without merging**